test/e2e/node_feature_discovery.go

```diff
@@ -153,263 +153,285 @@ func cleanupCRDs(cli *nfdclient.Clientset) {
+				Expect(err).NotTo(HaveOccurred())
 
-				fakeFeatureLabels := map[string]string{
-					nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature1": "true",
-					nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature2": "true",
-					nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3": "true",
-				}
+				// Remove pre-existing stale annotations and labels etc and CRDs
+				cleanupNode(f.ClientSet)
+				cleanupCRDs(nfdClient)
 
-				// Launch nfd-worker
-				By("Creating a nfd worker pod")
-				podSpecOpts := []testpod.SpecOption{
-					testpod.SpecWithRestartPolicy(corev1.RestartPolicyNever),
+				// Launch nfd-master
+				By("Creating nfd master pod and nfd-master service")
+				podSpecOpts := createPodSpecOpts(
 					testpod.SpecWithContainerImage(dockerImage),
-					testpod.SpecWithContainerExtraArgs("-oneshot", "-label-sources=fake"),
 					testpod.SpecWithTolerations(testTolerations),
-				}
-				workerPod := testpod.NFDWorker(podSpecOpts...)
-				workerPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), workerPod, metav1.CreateOptions{})
-				Expect(err).NotTo(HaveOccurred())
+					testpod.SpecWithContainerExtraArgs("-enable-taints"),
+				)
+				masterPod := e2epod.NewPodClient(f).CreateSync(testpod.NFDMaster(podSpecOpts...))
 
-				By("Waiting for the nfd-worker pod to succeed")
-				Expect(e2epod.WaitForPodSuccessInNamespace(f.ClientSet, workerPod.ObjectMeta.Name, f.Namespace.Name)).NotTo(HaveOccurred())
-				workerPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), workerPod.ObjectMeta.Name, metav1.GetOptions{})
+				// Create nfd-master service
+				nfdSvc, err := testutils.CreateService(f.ClientSet, f.Namespace.Name)
 				Expect(err).NotTo(HaveOccurred())
 
-				By(fmt.Sprintf("Making sure '%s' was decorated with the fake feature labels", workerPod.Spec.NodeName))
-				node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), workerPod.Spec.NodeName, metav1.GetOptions{})
-				Expect(err).NotTo(HaveOccurred())
-				for k, v := range fakeFeatureLabels {
-					Expect(node.Labels[k]).To(Equal(v))
-				}
-
-				// Check that there are no unexpected NFD labels
-				for k := range node.Labels {
-					if strings.HasPrefix(k, nfdv1alpha1.FeatureLabelNs) {
-						Expect(fakeFeatureLabels).Should(HaveKey(k))
-					}
-				}
+				By("Waiting for the nfd-master pod to be running")
+				Expect(e2epod.WaitTimeoutForPodRunningInNamespace(f.ClientSet, masterPod.Name, masterPod.Namespace, time.Minute)).NotTo(HaveOccurred())
 
-				By("Deleting the node-feature-discovery worker pod")
-				err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(context.TODO(), workerPod.ObjectMeta.Name, metav1.DeleteOptions{})
+				By("Verifying the node where nfd-master is running")
+				// Get updated masterPod object (we want to know where it was scheduled)
+				masterPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), masterPod.ObjectMeta.Name, metav1.GetOptions{})
```

Comment:
```suggestion
				masterPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), masterPod.Name, metav1.GetOptions{})
```
nit

Reply:
Thx! A bit unrelated to this PR as the first patch basically just indents the code by one tab (creating a function wrapper). All these ObjectMeta fixed separately in #1005

Reply:
cool, thanks